### PR TITLE
Fix failing exports when the project name contains invalid characters

### LINKED ISSUE #4171: Special characters in Asset name make exports fail
## Description 

Some characters are invalid when creating files (i.e. exports) on storage and should be stripped

## Additional info 

`\n` can be insert with `form_title` field in `settings` sheet in imported XLSForms.

### PR DESCRIPTION
**Pull request by noliveleger**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [x] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Newline and slashes are stripped from the name of the files when writing them to the storage.

## Related issues

Closes #4171

### CODE REVIEW

File: kpi/models/asset_file.py [hunk 1]
```diff
@@ -158,12 +159,15 @@ def get_download_url(self, request):
 
     @staticmethod
     def get_path(asset, file_type, filename):
+        default_storage = get_storage_class()()
+        valid_filename = default_storage.get_valid_name(filename)
```
Comment (jnm) [inline kpi/models/asset_file.py:163]: can `PRIVATE_STORAGE_CLASS` be something different from what's returned by this function?

File: kpi/tests/api/v2/test_api_assets.py [hunk 2]
```diff
@@ -1150,6 +1151,25 @@ def test_form_media_do_not_show_up_with_date_deleted_not_null(self):
         detail_response = self.client.get(detail_url)
         self.assertEqual(detail_response.status_code, status.HTTP_404_NOT_FOUND)
 
+    def test_upload_form_media_with_backslashes(self):
```
Comment (jnm) [inline kpi/tests/api/v2/test_api_assets.py:1154]: Is it a forward slash?

**Review by jnm — CHANGES_REQUESTED** [review]

File: kpi/tests/api/v2/test_api_assets.py (hunk 2 above)
Comment (noliveleger) [reply kpi/tests/api/v2/test_api_assets.py:1154]: Indeed ;-)

File: kpi/models/asset_file.py (hunk 1 above)
Comment (jnm) [reply kpi/models/asset_file.py:163]: idea:
```
In [15]: AssetFile._meta.get_field('content').storage
Out[15]: <private_storage.storage.files.PrivateFileSystemStorage at 0x7f8d053781c0>

In [16]: AssetFile._meta.get_field('content').storage.get_valid_name('ridic/\n\\ulous')
Out[16]: 'ridiculous'
```

**Review by jnm — APPROVED** [review]

**Pull request merged**